azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -901,3 +940,78 @@ def _process_cert_content(
+
+    def _process_cert_content(
+        self,
+        file_path: str,
+        file_name: str,
+        expected_exts: set[str],
+    ) -> Tuple[str, Union[x509.Certificate, x509.CertificateRevocationList]]:
+        """
+        Process the certificate content from the file, validate its extension,
+        content format and if certificate expired, and return the certificate
+        extension and decoded certificate object.
+        """
+        cert_extension = validate_file_extension(
+            file_name,
+            expected_exts,
+        )
+        # validate file content format by extension
+        expected_content_format = X509FileExtension.PEM.name if cert_extension == X509FileExtension.CRT.value\
+            else X509FileExtension.DER.name
+        certs = decode_x509_files(
+            read_file_content(file_path, read_as_binary=True),
+            expected_content_format,
+            cert_extension
+        )
+
+        if not certs:
+            raise InvalidArgumentValueError(
+                f"Error decoding certificate from file '{file_name}'. Please make sure the file is "
+                "a valid {expected_content_format} certificate."
+            )
+
+        # Only one certificate is expected in the PEM format.
+        if expected_content_format == X509FileExtension.PEM.name and len(certs) > 1:
+            raise InvalidArgumentValueError(
+                f"Multiple certificates detected in file '{file_name}' in {expected_content_format} format. "
+                f"Please provide a file with only one {expected_content_format} certificate."
+            )
+        cert = certs.pop()
+
+        # check for certificate expiry
+        if not cert_extension == X509FileExtension.CRL.value:
+            # check if the certificate is expired
+            expiry_date = cert.not_valid_after_utc
+            if expiry_date < datetime.now(timezone.utc):
+                raise InvalidArgumentValueError(
+                    f"Certificate in file '{file_name}' is expired. Please provide a valid certificate."
+                )
+
+        return cert_extension, cert
+
+    def _is_cert_self_signed(
+        self,
+        cert: x509.Certificate
+    ) -> bool:
+        # Check issuer and subject to determine if it's a self signed(non CA signed) certificate
+        issuer = cert.issuer
+        subject = cert.subject
+        return issuer == subject
+
+    def _is_ca_cert(
+        self,
+        cert: x509.Certificate,
+    ) -> bool:
+        # Check if it’s a CA cert
+        from cryptography.x509.oid import ExtensionOID
+        # this attribute only exist Version 3 of the X.509 standard
+        basic_constraints: x509.BasicConstraints = cert.extensions.get_extension_for_oid(
+            ExtensionOID.BASIC_CONSTRAINTS
+        ).value
+
+        if hasattr(basic_constraints, "ca"):
+            # if the certificate is a CA certificate
+            return basic_constraints.ca
```

Comment:
is `basic_constraints.ca` a bool (like the function return type indicates) that is being returned?

Reply:
yep!
<img width="500" alt="image" src="https://github.com/user-attachments/assets/8f23d676-abfb-432e-978d-4c327469dd3b" />